<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Creates an LD profile — create_LDprofile • zalpha</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Creates an LD profile — create_LDprofile" />
<meta property="og:description" content="An LD (linkage disequilibrium) profile is a look-up table containing the expected correlation between SNPs given the genetic distance between them. The use of an LD profile can increase the accuracy of results by taking into account the expected correlation between SNPs. This function aids the user in creating their own LD profile." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">zalpha</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0.9999</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/zalpha.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Creates an LD profile</h1>
    
    <div class="hidden name"><code>create_LDprofile.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>An LD (linkage disequilibrium) profile is a look-up table containing the expected correlation between SNPs given the genetic distance between them. The use of an LD profile can increase the accuracy of results by taking into account the expected correlation between SNPs. This function aids the user in creating their own LD profile.</p>
    </div>

    <pre class="usage"><span class='fu'>create_LDprofile</span><span class='op'>(</span><span class='va'>dist</span>, <span class='va'>x</span>, <span class='va'>bin_size</span>, max_dist <span class='op'>=</span> <span class='cn'>NULL</span>, beta_params <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dist</th>
      <td><p>A numeric vector, or a list of numeric vectors, containing the genetic distance for each SNP.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>A matrix of SNP values, or a list of matrices. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>dist</code> vector. SNPs should all be biallelic.</p></td>
    </tr>
    <tr>
      <th>bin_size</th>
      <td><p>The size of each bin, in the same units as <code>dist</code>.</p></td>
    </tr>
    <tr>
      <th>max_dist</th>
      <td><p>Optional. The maximum genetic distance to be considered. If this is not supplied, it will default to the maximum distance in the <code>dist</code> vector.</p></td>
    </tr>
    <tr>
      <th>beta_params</th>
      <td><p>Optional. Beta parameters are calculated if this is set to TRUE. Default is FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data frame containing an LD profile that can be used by other statistics in this package.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The input for <code>dist</code> and <code>x</code> can be lists. This allows multiple datasets to be used in the creation of the LD profile. For example, using all 22 autosomes from the human genome would involve 22 different distance vectors and SNP matrices.
Both lists should be the same length and should correspond exactly to each other (i.e. the distances in each element of <code>dist</code> should go with the SNPs in the same element of x)</p>
<p>In the output, bins represent lower bounds. The first bin contains pairs where the genetic distance is greater than or equal to 0 and less than <code>bin_size</code>. The final bin contains pairs where the genetic distance is greater than or equal to <code>max_dist</code>-<code>bin_size</code> and less than <code>max_dist</code>.
If the <code>max_dist</code> is not an increment of <code>bin_size</code>, it will be adjusted to the next highest increment. The final bin will be the bin that <code>max_dist</code> falls into. For example, if the <code>max_dist</code> is given as 4.5 and the <code>bin_size</code> is 1, the final bin will be 4.
<code>max_dist</code> should be big enough to cover the genetic distances between pairs of SNPs within the window size given when the \(Z_{\alpha}\) statistics are run. Any pairs with genetic distances bigger than <code>max_dist</code> will be assigned the values in the maximum bin of the LD profile.<br /></p>
<p>By default, Beta parameters are not calculated. To fit a Beta distribution to the expected correlations, needed for the <code><a href='Zalpha_BetaCDF.html'>Zalpha_BetaCDF</a></code> and <code><a href='Zbeta_BetaCDF.html'>Zbeta_BetaCDF</a></code> statistics, <code>beta_params</code> should be set to TRUE and the package 'fitdistrplus' must be installed.</p>
<p>Ideally, an LD profile would be generated using data from a null population with no selection, For example by using a simulation if the other population parameters are known. However, often these are unknown or complex, so generating an LD profile using the same data as is being analysed is acceptable, as long as the bins are large enough.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='Zalpha_expected.html'>Zalpha_expected</a></code>, <code><a href='Zalpha_rsq_over_expected.html'>Zalpha_rsq_over_expected</a></code>, <code><a href='Zalpha_log_rsq_over_expected.html'>Zalpha_log_rsq_over_expected</a></code>, <code><a href='Zalpha_Zscore.html'>Zalpha_Zscore</a></code>, <code><a href='Zalpha_BetaCDF.html'>Zalpha_BetaCDF</a></code>, <code><a href='Zbeta_expected.html'>Zbeta_expected</a></code>, <code><a href='Zbeta_rsq_over_expected.html'>Zbeta_rsq_over_expected</a></code>, <code><a href='Zbeta_log_rsq_over_expected.html'>Zbeta_log_rsq_over_expected</a></code>, <code><a href='Zbeta_Zscore.html'>Zbeta_Zscore</a></code>, <code><a href='Zbeta_BetaCDF.html'>Zbeta_BetaCDF</a></code>, <code><a href='Zalpha_all.html'>Zalpha_all</a></code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## load the snps example dataset</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>snps</span><span class='op'>)</span>
<span class='co'>## Create an LD profile using this data</span>
<span class='fu'>create_LDprofile</span><span class='op'>(</span><span class='va'>snps</span><span class='op'>$</span><span class='va'>cM_distances</span>,<span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>snps</span><span class='op'>[</span>,<span class='fl'>3</span><span class='op'>:</span><span class='fl'>12</span><span class='op'>]</span><span class='op'>)</span>,<span class='fl'>0.001</span><span class='op'>)</span>
</div><div class='output co'>#&gt;     bin        rsq        sd Beta_a Beta_b  n
#&gt; 1 0.000 0.10226664 0.1318628     NA     NA 59
#&gt; 2 0.001 0.14412346 0.1745857     NA     NA 51
#&gt; 3 0.002 0.09538328 0.1122312     NA     NA 41
#&gt; 4 0.003 0.11193736 0.1303776     NA     NA 28
#&gt; 5 0.004 0.19393939 0.1963924     NA     NA 11</div><div class='input'><span class='co'>## To get the Beta distribution parameter estimates, the fitdistrplus package is required</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"fitdistrplus"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>==</span><span class='cn'>TRUE</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='fu'>create_LDprofile</span><span class='op'>(</span><span class='va'>snps</span><span class='op'>$</span><span class='va'>cM_distances</span>,<span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='va'>snps</span><span class='op'>[</span>,<span class='fl'>3</span><span class='op'>:</span><span class='fl'>12</span><span class='op'>]</span><span class='op'>)</span>,<span class='fl'>0.001</span>,beta_params<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
<span class='op'>}</span>
</div><div class='output co'>#&gt; $start.arg
#&gt; $start.arg$shape1
#&gt; [1] 0.5319322
#&gt; 
#&gt; $start.arg$shape2
#&gt; [1] 4.347826
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; 
#&gt; $start.arg
#&gt; $start.arg$shape1
#&gt; [1] 0.5237006
#&gt; 
#&gt; $start.arg$shape2
#&gt; [1] 2.942187
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; 
#&gt; $start.arg
#&gt; $start.arg$shape1
#&gt; [1] 0.7421819
#&gt; 
#&gt; $start.arg$shape2
#&gt; [1] 6.309295
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; 
#&gt; $start.arg
#&gt; $start.arg$shape1
#&gt; [1] 0.7818732
#&gt; 
#&gt; $start.arg$shape2
#&gt; [1] 5.433496
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; 
#&gt; $start.arg
#&gt; $start.arg$shape1
#&gt; [1] 1.098982
#&gt; 
#&gt; $start.arg$shape2
#&gt; [1] 3.856677
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; </div><div class='output co'>#&gt;     bin        rsq        sd    Beta_a   Beta_b  n
#&gt; 1 0.000 0.10226664 0.1318628 0.8232427 6.435184 59
#&gt; 2 0.001 0.14412346 0.1745857 0.7323277 3.960080 51
#&gt; 3 0.002 0.09538328 0.1122312 0.9962786 8.322913 41
#&gt; 4 0.003 0.11193736 0.1303776 1.0432924 7.113517 28
#&gt; 5 0.004 0.19393939 0.1963924 1.4733393 4.979831 11</div><div class='input'>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Clare Horscroft.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


